plugins/source/aws/codegen/recipes/base.go

```diff
@@ -28,17 +28,24 @@ type Resource struct {
 	SkipFields            []string
 	Description           string
 	ExtraColumns          []codegen.ColumnDefinition
+	PKColumns             []string
 	Table                 *codegen.TableDefinition
 	Multiplex             string
 	PreResourceResolver   string
 	PostResourceResolver  string
 	Relations             []string
 	UnwrapEmbeddedStructs bool
 
+	// NameTransformer custom name transformer for resource
+	NameTransformer func(field reflect.StructField) (string, error)
```

Comment:
I'm aware we have the same thing in the GCP plugin, but it seems a bit like overkill to have a generic transformer function here, when (I'm pretty sure) `map[string]string` would suffice for all cases we know of in AWS 😅 The use case in GCP is also slightly different: it's for overriding things like "CORS" with "cors" instead of "c_o_r_s" (and I think we actually have better mechanisms for handling this now)

Could we maybe make it something like `OverrideColumnNames: map[string]string{}` with the key being the struct field, and the value being the new column name? So then generally it would be used something like:

```
OverrideColumnNames: map[string]string{
    "ResourceShareArn": "arn",
}
```

This also has the advantage that you don't need to guess what the transformed column name will be, you can use the struct field name, which is certain.

What do you think?

Reply:
If we consider the field names as keys, only the name transformer option can help us at the moment.
We can do a simple map if we use the resulting column names instead, but for now I think it works better.